spec/FrameSpec.js

```diff
@@ -0,0 +1,73 @@
+describe("Frame", function() {
```

Comment:
Really solid testing overall! Though maybe the use of Jasmine spies could of helped, however, that might be something we can discuss, as I am not fully sure on how to use them myself!